src/Stripe.net/Services/Invoices/InvoiceUpcomingListInvoiceLineItemsOptions.cs

```diff
@@ -0,0 +1,72 @@
+namespace Stripe
+{
+    using System;
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using Stripe.Infrastructure;
+
+    public class InvoiceUpcomingListInvoiceLineItemsOptions : ListOptions
+    {
+        [JsonProperty("coupon")]
+        public string Coupon { get; set; }
+
+        [JsonProperty("customer")]
+        public string Customer { get; set; }
+
+        [JsonProperty("discounts")]
+        public List<InvoiceLineItemDiscountOptions> Discounts { get; set; }
```

Comment:
So this name bothers me for two reasons.

First, it could conflict with other options on the Create/Update path. That's the "problem" with upcoming because it has its own set of parameters/hashes but because we "share" everything that is a nested hash they can conflict. It's definitely not the end of the world and likely the right decision but damn this is definitely confusing.

Second, this is an option on Upcoming invoice, but you name it as if it's an option on InvoiceLineItem which _feels_ incorrect. Shouldn't it be `<parent><child>Options`? Like if we shipped a `createPerson` on `AccountService`, wouldn't you call the option `AccountPersonCreateOptions` or `AccountCreatePersonOptions`?